src/query/functions/temporal/aggregation.go

```diff
@@ -0,0 +1,196 @@
+
+	// CountTemporalType calculates count of all values in the specified interval
+	CountTemporalType = "count_over_time"
+
+	// MinTemporalType calculates the minimum of all values in the specified interval
+	MinTemporalType = "min_over_time"
+
+	// MaxTemporalType calculates the maximum of all values in the specified interval
+	MaxTemporalType = "max_over_time"
+
+	// SumTemporalType calculates the sum of all values in the specified interval
+	SumTemporalType = "sum_over_time"
+
+	// StdDevTemporalType calculates the standard deviation of all values in the specified interval
+	StdDevTemporalType = "stddev_over_time"
+
+	// StdVarTemporalType calculates the standard variance of all values in the specified interval
+	StdVarTemporalType = "stdvar_over_time"
+)
+
+var (
+	aggFuncs = map[string]func([]float64) float64{
+		AvgTemporalType:    avgOverTime,
+		CountTemporalType:  countOverTime,
+		MinTemporalType:    minOverTime,
+		MaxTemporalType:    maxOverTime,
+		SumTemporalType:    sumOverTime,
+		StdDevTemporalType: stddevOverTime,
+		StdVarTemporalType: stdvarOverTime,
+	}
+)
+
+// NewAggOp creates a new base temporal transform with a specified node
+func NewAggOp(args []interface{}, optype string) (transform.Params, error) {
+	if _, ok := aggFuncs[optype]; !ok {
+		return emptyOp, fmt.Errorf("unknown aggregation type: %s", optype)
+	}
+
+	return newBaseOp(args, optype, newAggNode)
+}
+
+func newAggNode(op baseOp, controller *transform.Controller) Processor {
+	return &aggNode{
+		op:         op,
+		controller: controller,
+		aggFunc:    aggFuncs[op.operatorType],
+	}
+}
+
+type aggNode struct {
+	op         baseOp
+	controller *transform.Controller
+	aggFunc    func([]float64) float64
+}
+
+func (a *aggNode) Process(values []float64) float64 {
+	return a.aggFunc(values)
+}
+
+func avgOverTime(values []float64) float64 {
+	var sum, count float64
+	for _, v := range values {
+		if !math.IsNaN(v) {
+			count++
+			sum += v
+		}
+	}
+
+	if count == 0 {
+		return math.NaN()
+	}
+	return sum / count
+}
+
+func countOverTime(values []float64) float64 {
+	var count float64
+	for _, v := range values {
+		if !math.IsNaN(v) {
+			count++
+		}
+	}
+	return count
+}
+
+func minOverTime(values []float64) float64 {
+	var count float64
+	min := math.Inf(1)
+	for _, v := range values {
+		if !math.IsNaN(v) {
+			count++
+			min = math.Min(min, v)
+		}
+	}
+
+	if count == 0 {
+		return math.NaN()
+	}
+	return min
+}
+
+func maxOverTime(values []float64) float64 {
+	var count float64
+	max := math.Inf(-1)
+	for _, v := range values {
+		if !math.IsNaN(v) {
+			count++
+			max = math.Max(max, v)
+		}
+	}
+
+	if count == 0 {
+		return math.NaN()
+	}
+	return max
+}
+
+func sumOverTime(values []float64) float64 {
+	var sum, count float64
+	for _, v := range values {
+		if !math.IsNaN(v) {
+			count++
+			sum += v
+		}
+	}
+
+	if count == 0 {
+		return math.NaN()
+	}
+	return sum
+}
+
+func stddevOverTime(values []float64) float64 {
```

Comment:
instead of implementing all these functions, can you use : gonum ? Eg: https://github.com/gonum/gonum/blob/73ea1e732937f96d723d31dc5263d214a275d204/stat/stat.go#L1199